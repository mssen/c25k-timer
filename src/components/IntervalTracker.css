.current-interval {
  font-size: 100px;
}

.total-interval {
  font-size: 40px;
  color: gray;
}