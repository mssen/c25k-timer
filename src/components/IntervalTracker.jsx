import React from 'react';
import PropTypes from 'prop-types';

import './IntervalTracker.css';

const IntervalTracker = ({ current, total }) => {
  const displayCurrent = current > total ? current - 1 : current;

  return (
    <div>
      <span className="current-interval">{ displayCurrent }</span>
      /
      <span className="total-interval">{ total }</span>
    </div>
  )
};

IntervalTracker.propTypes = {
  current: PropTypes.number.isRequired,
  total: PropTypes.number.isRequired
}

export default IntervalTracker;