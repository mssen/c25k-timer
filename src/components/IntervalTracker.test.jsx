import React from 'react';
import { shallow } from 'enzyme';
import IntervalTracker from './IntervalTracker';

it('renders without crashing', () => {
  shallow(<IntervalTracker current={0} total={0} />);
});

it('shows total / total when current interval is more than total', () => {
  const wrapper = shallow(<IntervalTracker current={5} total={4} />);
  expect(wrapper).toMatchSnapshot();
});

it('shows current interval', () => {
  const wrapper = shallow(<IntervalTracker current={1} total={4} />);
  expect(wrapper).toMatchSnapshot();
});
