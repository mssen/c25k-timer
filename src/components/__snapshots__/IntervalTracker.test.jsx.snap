// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shows current interval 1`] = `
<div>
  <span
    className="current-interval"
  >
    1
  </span>
  /
  <span
    className="total-interval"
  >
    4
  </span>
</div>
`;

exports[`shows total / total when current interval is more than total 1`] = `
<div>
  <span
    className="current-interval"
  >
    4
  </span>
  /
  <span
    className="total-interval"
  >
    4
  </span>
</div>
`;
